lib/core/include/scipp/core/spatial_transforms.h

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#pragma once
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+#include "scipp/core/dtype.h"
+
+namespace scipp::core {
+
+class RotationTransform {
+private:
+  // Store as quaterniond as this is more space efficient than storing as matrix
+  // (4 doubles for quat vs 9 doubles for 3x3 matrix).
+  Eigen::Quaterniond quat;
+
+public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+  RotationTransform() : quat(Eigen::Quaterniond::Identity()){};
+  explicit RotationTransform(const Eigen::Quaterniond &x) : quat(x){};
+
+  [[nodiscard]] Eigen::Matrix3d matrix() const {
+    return quat.toRotationMatrix();
+  }
+
+  inline bool operator==(const RotationTransform &other) const {
+    return this->matrix() == other.matrix();
+  }
+};
+
+class ScalingTransform {
+private:
+  Eigen::DiagonalMatrix<double, 3> mat;
+
+public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+  ScalingTransform()
+      : mat(Eigen::Matrix3d::Identity().diagonal().asDiagonal()){};
+  explicit ScalingTransform(const Eigen::DiagonalMatrix<double, 3> &x)
+      : mat(x){};
+
+  [[nodiscard]] Eigen::Matrix3d matrix() const { return mat; }
+
+  inline bool operator==(const ScalingTransform &other) const {
+    return this->matrix() == other.matrix();
+  }
+};
+
+class TranslationTransform {
+private:
+  Eigen::Vector3d vec;
```

Comment:
Any reason why this is not using `Eigen::Translation`?

Reply:
The [eigen docs](https://eigen.tuxfamily.org/dox/classEigen_1_1Translation.html) specifically recommend against using `Eigen::Translation` to *store* a translation. To quote from the docs:

> This class is not aimed to be used to store a translation transformation, but rather to make easier the constructions and updates of Transform objects.